src/os/CMakeLists.txt

```diff
@@ -32,6 +32,7 @@ if(WITH_BLUESTORE)
     bluestore/FreelistManager.cc
     bluestore/StupidAllocator.cc
     bluestore/BitmapAllocator.cc
+    bluestore/io_uring.cc
```

Comment:
i think we should compile this source file only if `WITH_LIBURING`. like
```cmake
if(WITH_LIBURING)
  list(APPEND libos_srcs bluestore/io_uring.cc)
  # add liburing as an external project and link against it
endif(WITH_LIBURING)
```

Reply:
No, the trick is the following: we always include io_uring.cc, but if HAVE_LIBURING is not defined - bool ioring_queue_t::supported() returns false.  (please see io_uring.cc, bottom of the file).

Why? Because then we do not need any ifdefs in KernelDevice.cc,  see the following chunk:
```

  bool use_ioring = g_ceph_context->_conf.get_val<bool>("bluestore_ioring");
  unsigned int iodepth = cct->_conf->bdev_aio_max_queue_depth;

  if (use_ioring && ioring_queue_t::supported()) {
    io_queue = std::unique_ptr<io_queue_t>(new ioring_queue_t(iodepth));
  }
```

Thanks for the rest comments.  I will refactor that.